File: Solution Using C#/Fawry.Ecommerce/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fawry.Ecommerce
{
    internal class Customer
    {
        public double Balance { get; set; }
        public Cart Cart { get; }
        public Customer(double Balance)
        {
            this.Balance = Balance;
            Cart = new Cart();
        }
        public void AddToCart(Product product, int quantity)
        {
            Cart.AddItem(product, quantity);
        }
        
    }
}

File: Solution Using Java/src/models/Customer.java
package models;

public class Customer {
    private double balance;
    private final Cart cart;

    public Customer(double balance) {
        this.balance = balance;
        this.cart = new Cart();
    }
    public double getBalance() {
        return balance;
    }
    public void setBalance(double balance) {
        this.balance = balance;
    }
    public Cart getCart() {
        return cart;
    }
    public void addToCart(Product product,int quantity) {
        cart.addItem(product,quantity);
    }
}
